cns/service/main.go

```diff
@@ -1303,6 +1303,68 @@ func InitializeCRDState(ctx context.Context, httpRestService cns.HTTPService, cn
 	// TODO(rbtr): nodename and namespace should be in the cns config
 	directscopedcli := nncctrl.NewScopedClient(directnnccli, types.NamespacedName{Namespace: "kube-system", Name: nodeName})
 
+	// Create the base NNC CRD if HomeAz is enabled
+	if cnsconfig.EnableHomeAZ {
```

Comment:
better to extract the scope opened by this flag to a different function. it will help clean up some code paths.